src/factory-in-proc.js

```diff
@@ -66,9 +66,13 @@ class FactoryInProc {
       options = {}
     }
     const IPFS = this.exec
-    new IPFS(options).version((err, _version) => {
-      if (err) { callback(err) }
-      callback(null, _version)
+    const node = new IPFS(options)
+
+    node.once('ready', () => {
+      node.version((err, _version) => {
+        if (err) { callback(err) }
```

Comment:
needs return.

Reply:
Ha! Thanks, you really have an eye for these. Perhaps we could write a lint warning for this 😉?

Reply:
@diasdavid Looks like there already is an ESLint rule for this: https://eslint.org/docs/rules/callback-return

Adding a lint rule to aegir would be a big change across our projects but it might be worth it for this one as it's not about style but functionality and could catch bugs. It might be okay to set it with error severity but we would want to pre-run some repos to check.

We could alternatively phase it in as a warning and then bump to an error in a month or two, only problem is we don't really pay attention to warnings. If it ends up finding a lot of cases we could do a fix chore of key repos during that period.

Reply:
I say enable the lint rule in aegir as an error and see what happens.